chain/chain/src/test_utils.rs

```diff
@@ -1034,12 +1034,33 @@ impl RuntimeAdapter for KeyValueRuntime {
 
     fn get_gc_stop_height(&self, block_hash: &CryptoHash) -> BlockHeight {
         if !self.no_gc {
+            // This code is 'incorrect' - as production one is always setting the GC to the
+            // first block of the epoch.
+            // Unfortunately many tests are depending on this and not setting epochs when
+            // they produce blocks.
             let block_height = self
                 .get_block_header(block_hash)
                 .unwrap_or_default()
                 .map(|h| h.height())
                 .unwrap_or_default();
             block_height.saturating_sub(NUM_EPOCHS_TO_KEEP_STORE_DATA * self.epoch_length)
+        /*  // TODO: use this version of the code instead - after we fix the block creation
```

Comment:
@mm-near Is there an issue that tracks this TODO

Reply:
good catch. Created CP-35